pkg/apis/marketplace/v1alpha1/catalogsourceconfig_types.go

```diff
@@ -13,27 +14,47 @@ const (
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// CatalogSourceConfigList contains a list of CatalogSourceConfig
 type CatalogSourceConfigList struct {
 	metav1.TypeMeta `json:",inline"`
-	metav1.ListMeta `json:"metadata"`
+	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []CatalogSourceConfig `json:"items"`
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// CatalogSourceConfig is the Schema for the catalogsourceconfigs API
+// +k8s:openapi-gen=true
 type CatalogSourceConfig struct {
 	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata"`
-	Spec              CatalogSourceConfigSpec   `json:"spec"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              CatalogSourceConfigSpec   `json:"spec,omitempty"`
 	Status            CatalogSourceConfigStatus `json:"status,omitempty"`
 }
 
+// CatalogSourceConfigSpec defines the desired state of CatalogSourceConfig
 type CatalogSourceConfigSpec struct {
 	TargetNamespace string `json:"targetNamespace"`
 	Packages        string `json:"packages"`
 }
 
+// CatalogSourceConfigStatus defines the observed state of CatalogSourceConfig
 type CatalogSourceConfigStatus struct {
 	// Current phase of the CatalogSourceConfig object.
 	CurrentPhase ObjectPhase `json:"currentPhase,omitempty"`
 }
+
+func init() {
+	SchemeBuilder.Register(&CatalogSourceConfig{}, &CatalogSourceConfigList{})
+}
+
+// Set group, version, and kind strings from reference
+// Object the sdk client returns does not set these
```

Comment:
I don't follow this comment. What `reference` are you talking about here? Or are you trying to say "Set group, version, and kind strings from **_the_** reference object **_as_** the sdk client returns does not set these"?

Reply:
I'm not sure I follow your message. Parenthetical included:

Set group, version, and kind strings from **(the internal)** reference **(that we defined in the v1alpha1 package)**
**(The)** Object the sdk client returns does not set these **(so we must find the correct values and set them manually)**

Is that more clear?

Reply:
Yes, that makes it much clearer.